# Integrated Multi-Omics Reveal Polycomb Repressive Complex 2 Restricts Human Trophoblast Induction

## Code availability
Scripts related to the manuscript entitled "Integrated Multi-Omics Reveal Polycomb Repressive Complex 2 Restricts Human Trophoblast Induction" published in ...(https://www.) are available in this GitHub repository in ... directory.

## Loom files for scRNAseq data visualization
As a resource to community we provide .loom files that contain UMAP-clustering, cluster annotation and gene expression from the scRNA-seq datasets analyzed in the manuscript. 
.loom files can be downloaded here: [GSE176173](https://www.ncbi.nlm.nih.gov/geo/...) using the http links at the bottom of the page. After unzipping, you can browse interactively using [SCope platform](https://scope.aertslab.org). SCope was developed and is maintained by [Aerts lab.](https://www.aertslab.org/).
In order to browse the datasets, upload loom files by clicking on "Upload dataset" in the menu on the left-side of SCope website. Uploading may take a few minutes. Successful upload will activate a private, temporary session for the user. Each session expires after 5 days, but can be renewed by re-uploading the loom file. 
Standalone SCope apps for macOS and Linux can be downloaded from [here](https://github.com/aertslab/SCope/releases).

## Shiny App for multi-omic data exploration
The multi-omic data can be explored using the online resource: https://www.bioinformatics.babraham.ac.uk/shiny/shiny_omics/Shiny_omics (Username: test; Password: justtesting).

## Data availability
Raw and processed sequencing data for  cCUT&RUN, RNA-seq and scRNA-seq can be found here: [GSE176175](http://www.ncbi.nlm.nih.gov/geo/...).

The ChEP mass spectrometry proteomics data can be found here: [](https://www.ebi.ac.uk/pride/...).

The hPTM mass spectrometry proteomics data can be found here: [](https://www.ebi.ac.uk/pride/...).

Additional raw mass spectrometry files are available here: [](http://chorusproject.org/...).


